### PR TITLE
Use OTEL_DOTNET_AUTO_ as the env var prefix 

### LINKED ISSUE #293: Use `OTEL_DOTNET_AUTO_` as prefix for env vars introduced in this repository
Motivation: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#language-specific-environment-variables

_Originally posted by @pjanotti in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/288#discussion_r763480106_

### PR DESCRIPTION
**Pull request by pellared**
Fixes #293

Changes proposed in this pull request:

- Remove support for OTEL_{0}_ENABLED env var
- "Rename" the following env vars to have `OTEL_DOTNET_AUTO` prefix
   - OTEL_DOTNET_TRACER_HOME
   - OTEL_TRACE_ENABLED
   - OTEL_PROFILER_PROCESSES
   - OTEL_PROFILER_EXCLUDE_PROCESSES
   - OTEL_AZURE_APP_SERVICES
   - OTEL_INTEGRATIONS
   - OTEL_DOTNET_TRACER_INSTRUMENTATIONS
   - OTEL_DOTNET_TRACER_DISABLED_INSTRUMENTATIONS
   - OTEL_TRACE_{0}_ENABLED
   - OTEL_TRACE_DOMAIN_NEUTRAL_INSTRUMENTATION
   - OTEL_CLR_DISABLE_OPTIMIZATIONS
   - OTEL_CLR_ENABLE_INLINING
   - OTEL_CLR_ENABLE_NGEN
   - OTEL_TRACE_LOG_DIRECTORY
   - OTEL_TRACE_LOG_PATH
   - OTEL_TRACE_DEBUG
   - OTEL_DOTNET_TRACER_CONSOLE_EXPORTER_ENABLED
   - OTEL_DUMP_ILREWRITE_ENABLED
   - OTEL_DOTNET_TRACER_LOAD_AT_STARTUP
   - OTEL_DOTNET_TRACER_ADDITIONAL_SOURCES
   - OTEL_DOTNET_TRACER_LEGACY_SOURCES
   - OTEL_DOTNET_TRACER_INSTRUMENTATION_PLUGINS
   - OTEL_INTERNAL_TRACE_NATIVE_ENGINE_PATH- 
- Document `OTEL_DOTNET_AUTO_HOME`

### CODE REVIEW

File: docs/USAGE.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 | Environment variable | Description | Default |
 |-|-|-|
+| `OTEL_DOTNET_TRACER_HOME` | Installation location. | `true` |
```
[the remark this answers is not in the document]
Comment (pellared) [reply docs/USAGE.md:11]: in progress 😄

File: src/OpenTelemetry.ClrProfiler.Managed/Configuration/IntegrationSettings.cs [hunk 2]
```diff
@@ -21,8 +21,7 @@ public IntegrationSettings(string integrationName, IConfigurationSource source)
                 return;
             }
 
-            Enabled = source.GetBool(string.Format(ConfigurationKeys.Integrations.Enabled, integrationName)) ??
-                      source.GetBool(string.Format("OTEL_{0}_ENABLED", integrationName));
+            Enabled = source.GetBool(string.Format(ConfigurationKeys.Integrations.Enabled, integrationName));
```
Comment (nrcventura) [inline src/OpenTelemetry.ClrProfiler.Managed/Configuration/IntegrationSettings.cs:24]: Did we intentionally remove the fallback setting check? If so it's not clear why we're removing it as part of this PR.

File: src/OpenTelemetry.ClrProfiler.Native/environment_variables.h [hunk 3]
```diff
@@ -77,17 +77,17 @@ const WSTRING azure_app_services_cli_telemetry_profile_value =
...
 // Enable the profiler to dump the IL original code and modification to the log.
-const WSTRING dump_il_rewrite_enabled = WStr("OTEL_DUMP_ILREWRITE_ENABLED");
+const WSTRING dump_il_rewrite_enabled = WStr("OTEL_DOTNET_AUTO_DUMP_ILREWRITE_ENABLED");
 
 // Sets whether to enable JIT inlining
-const WSTRING clr_enable_inlining = WStr("OTEL_CLR_ENABLE_INLINING");
+const WSTRING clr_enable_inlining = WStr("OTEL_DOTNET_AUTO_CLR_ENABLE_INLINING");
 
 // Custom internal tracer profiler path
 const WSTRING internal_trace_profiler_path =
```
Comment (zacharycmontoya) [inline src/OpenTelemetry.ClrProfiler.Native/environment_variables.h:86]: BTW in a later PR this specific variable can be removed, it's not used anywhere in this repo

Comment (zacharycmontoya) [reply src/OpenTelemetry.ClrProfiler.Native/environment_variables.h:86]: Sorry, I was referring to `internal_trace_profiler_path`, not `clr_enable_inlining`

File: src/OpenTelemetry.ClrProfiler.Managed/Configuration/IntegrationSettings.cs (hunk 2 above)
Comment (zacharycmontoya) [reply src/OpenTelemetry.ClrProfiler.Managed/Configuration/IntegrationSettings.cs:24]: When reading Chris's comment I was mixing this up with `OTEL_DOTNET_AUTO_{0}_ENABLED` (previously `OTEL_TRACE_{0}_ENABLED`). While a pure refactoring PR would be nice, it does make sense to remove the second undocumented key so I'm OK with including this in the PR

**Review by nrcventura — APPROVED** [review]

File: src/OpenTelemetry.ClrProfiler.Native/environment_variables.h (hunk 3 above)
Comment (pjanotti) [reply src/OpenTelemetry.ClrProfiler.Native/environment_variables.h:86]: Let's do it in a separate PR.

**Review by pjanotti — APPROVED** [review]
LGTM

File: src/OpenTelemetry.ClrProfiler.Native/environment_variables.h (hunk 3 above)
Comment (pellared) [reply src/OpenTelemetry.ClrProfiler.Native/environment_variables.h:86]: Issue created https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/360

**Pull request merged**